lib/mail_room/mailbox_handler.rb

```diff
@@ -45,17 +43,19 @@ def new_messages
     # search for all new (unseen) message ids
     # @return [Array<Integer>] message ids
     def new_message_ids
-      @imap.search(@mailbox.search_command)
+      uids = @imap.uid_search(@mailbox.search_command)
```

Comment:
Yeah, added that in to be able to add some debugging puts statements between the two calls :)